walkthroughs/cdk-examples/howto-mutual-tls-file-provided-by-acm/lib/stacks/mesh-components.ts

```diff
@@ -0,0 +1,162 @@
+      console.log("\n\n -------------------------------------------------------- \n\n");
+    }
+
+    this.virtualGateway = new appmesh.VirtualGateway(this, `${this.stackName}VirtualGateway`, {
+      mesh: this.mesh,
+      virtualGatewayName: "ColorGateway",
+      listeners: [appmesh.VirtualGatewayListener.http({ port: this.serviceDiscovery.infra.port })],
+      backendDefaults: this.buildGatewayTls(meshUpdateChoice, props),
+    });
+
+    this.virtualNode = new appmesh.VirtualNode(this, `${this.stackName}ColorTellerVn`, {
+      mesh: this.mesh,
+      virtualNodeName: "color-teller",
+      serviceDiscovery: this.serviceDiscovery.getAppMeshServiceDiscovery(this.serviceDiscovery.infra.serviceColorTeller),
+      listeners: [
+        appmesh.VirtualNodeListener.http({
+          port: this.serviceDiscovery.infra.port,
+          tls: this.buildVirtualNodeTls(meshUpdateChoice, props),
+        }),
+      ],
+    });
+
+    this.virtualGateway.node.addDependency(this.virtualNode);
+
+    this.virtualService = new appmesh.VirtualService(this, `${this.stackName}VirtualService`, {
+      virtualServiceName: `colorteller.${this.node.tryGetContext("SERVICES_DOMAIN")}`,
+      virtualServiceProvider: appmesh.VirtualServiceProvider.virtualNode(this.virtualNode),
+    });
+
+    this.virtualGateway.addGatewayRoute(`${this.stackName}VirtualGatewayRoute`, {
+      gatewayRouteName: "gateway-gr",
+      routeSpec: appmesh.GatewayRouteSpec.http({
+        routeTarget: this.virtualService,
+      }),
+    });
+
+    this.updateServicesRole = new iam.Role(this, `${this.stackName}UpdateSvcsRole`, {
+      assumedBy: new iam.ServicePrincipal("lambda.amazonaws.com"),
+      managedPolicies: addManagedPolicies(this, "updateScvPols", "AmazonECS_FullAccess", "service-role/AWSLambdaBasicExecutionRole"),
+    });
+
+    this.updateServicesFunc = new lambda.DockerImageFunction(this, `${this.stackName}InitCertFunc`, {
+      functionName: "update-ecs-services",
+      logRetention: logs.RetentionDays.ONE_DAY,
+      timeout: Duration.seconds(900),
+      code: lambda.DockerImageCode.fromImageAsset(path.join(__dirname, "../../lambdas/"), {
+        platform: assets.Platform.LINUX_AMD64,
+        cmd: ["updateservices.lambda_handler"],
+        buildArgs: { FILE: "updateservices.py" },
+      }),
+      role: this.updateServicesRole,
+      environment: {
+        SVC_GATEWAY: this.serviceDiscovery.infra.serviceGateway,
+        SVC_TELLER: this.serviceDiscovery.infra.serviceColorTeller,
+        CLUSTER: this.serviceDiscovery.infra.cluster.clusterName,
+        MESH_UPDATE: meshUpdateChoice,
+      },
+    });
+
+    this.updateServicesTrigger = new triggers.Trigger(this, `${this.stackName}UpdateSvcTrg`, {
+      handler: this.updateServicesFunc,
+      executeAfter: [this.virtualGateway],
+      executeOnHandlerChange: false,
+    });
+
+    // https://github.com/aws/aws-cdk/issues/19272#issuecomment-1091799069
```

Comment:
Add a bit wording here instead of just the link.

Reply:
Noted